packages/url_launcher/url_launcher_ios/example/ios/RunnerTests/URLLauncherTests.swift

```diff
@@ -131,6 +131,24 @@ final class URLLauncherTests: XCTestCase {
     XCTAssertEqual(launcher.passedOptions?[.universalLinksOnly] as? Bool, true)
   }
 
+  func testLaunchSafariViewControllerWithClose() {
+    let launcher = FakeLauncher()
+    let plugin = createPlugin(launcher: launcher)
+
+    let expectation = XCTestExpectation(description: "completion called")
+    plugin.openUrlInSafariViewController(url: "https://flutter.dev") { result in
```

Comment:
I worry that this test would flake if `https://flutter.dev` loads before the test calls `closeSafariViewController`. Is there some URL we could test that is known to hang forever? Or am I overthinking this? cc @stuartmorgan

Reply:
I would not expect it to be possible for that to happen. Unless `SFSafariViewController`'s `didCompleteInitialLoad` callback happens on a non-main thread (which I wouldn't expect since it passed the view controller as a parameter), it shouldn't be possible for the callback to trigger until the runloop runs again, which I would think could not happen before the `wait` call.

There are some assumptions in there, so I wouldn't be confident in saying that it's impossible, but I'd be fine with waiting to see if it does actually happen before we add a bunch of extra logic (since the reliable way to avoid that would be dependency injection of a fake `SFSafariViewController` that we know won't load anything) to avoid I case that I strongly suspect is in fact impossible.